lib/dynflow/rails/daemon.rb

```diff
@@ -1,32 +1,54 @@
 require 'fileutils'
+require 'daemons'
+require 'get_process_mem'
+require 'dynflow/watchers/memory_consumption_watcher'
 
 module Dynflow
   class Rails
     class Daemon
+      attr_reader :dynflow_memory_watcher_class, :daemons_class
+
+      # make Daemon dependency injection ready for testing purposes
+      def initialize(
+        dynflow_memory_watcher_class = ::Dynflow::Watchers::MemoryConsumptionWatcher,
+        daemons_class = ::Daemons
```

Comment:
We should set to nil here, and create a default inside the `daemons_class` accessor. This way it would be accessed only on demand (from `run_background`).